### ISSUE TITLE
segmentation fault

### ISSUE THREAD

**Issue by jigyasa06** [tensorflow/nmt#350]
Jigyasas-MBP:nmt jigyasa$ python -m nmt.nmt --attention=scaled_luong --src=de --tgt=bo --vocab_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/vocab --train_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/train --dev_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/validate --test_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/test --out_dir=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/tmp/nmt_model --num_train_steps=10000 --steps_per_stats=100 --num_layers=2 --num_units=128 --dropout=0.2 --metrics=bleu --batch_size=28
/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
# Job id 0
# hparams:
  src=de
  tgt=bo
  train_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/train
  dev_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/validate
  test_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/test
  out_dir=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/tmp/nmt_model
# Vocab file /Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/vocab.de exists
# Vocab file /Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/vocab.bo exists
  saving hparams to /Users/jigyasa/Desktop/nmt_jigyasa/nmt/tmp/nmt_model/hparams
  saving hparams to /Users/jigyasa/Desktop/nmt_jigyasa/nmt/tmp/nmt_model/best_bleu/hparams
  attention=scaled_luong
  attention_architecture=standard
  avg_ckpts=False
  batch_size=28
  beam_width=0
  best_bleu=0
  best_bleu_dir=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/tmp/nmt_model/best_bleu
  check_special_token=True
  colocate_gradients_with_ops=True
  decay_scheme=
  dev_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/validate
  dropout=0.2
  embed_prefix=None
  encoder_type=uni
  eos=</s>
  epoch_step=0
  forget_bias=1.0
  infer_batch_size=32
  init_op=uniform
  init_weight=0.1
  learning_rate=1.0
  length_penalty_weight=0.0
  log_device_placement=False
  max_gradient_norm=5.0
  max_train=0
  metrics=['bleu']
  num_buckets=5
  num_decoder_layers=2
  num_decoder_residual_layers=0
  num_embeddings_partitions=0
  num_encoder_layers=2
  num_encoder_residual_layers=0
  num_gpus=1
  num_inter_threads=0
  num_intra_threads=0
  num_keep_ckpts=5
  num_layers=2
  num_train_steps=10000
  num_translations_per_input=1
  num_units=128
  optimizer=sgd
  out_dir=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/tmp/nmt_model
  output_attention=True
  override_loaded_hparams=False
  pass_hidden_state=True
  random_seed=None
  residual=False
  sampling_temperature=0.0
  share_vocab=False
  sos=<s>
  src=de
  src_embed_file=
  src_max_len=50
  src_max_len_infer=None
  src_vocab_file=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/vocab.de
  src_vocab_size=11178
  steps_per_external_eval=None
  steps_per_stats=100
  subword_option=
  test_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/test
  tgt=bo
  tgt_embed_file=
  tgt_max_len=50
  tgt_max_len_infer=None
  tgt_vocab_file=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/vocab.bo
  tgt_vocab_size=9987
  time_major=True
  train_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/train
  unit_type=lstm
  vocab_prefix=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/nmt/scripts/iwslt15/vocab
  warmup_scheme=t2t
  warmup_steps=0
# creating train graph ...
  num_layers = 2, num_residual_layers=0
  cell 0  LSTM, forget_bias=1  DropoutWrapper, dropout=0.2   DeviceWrapper, device=/gpu:0
  cell 1  LSTM, forget_bias=1  DropoutWrapper, dropout=0.2   DeviceWrapper, device=/gpu:0
  cell 0  LSTM, forget_bias=1  DropoutWrapper, dropout=0.2   DeviceWrapper, device=/gpu:0
  cell 1  LSTM, forget_bias=1  DropoutWrapper, dropout=0.2   DeviceWrapper, device=/gpu:0
  learning_rate=1, warmup_steps=0, warmup_scheme=t2t
  decay_scheme=, start_decay_step=10000, decay_steps 0, decay_factor 1
# Trainable variables
  embeddings/encoder/embedding_encoder:0, (11178, 128), /device:GPU:0
  embeddings/decoder/embedding_decoder:0, (9987, 128), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_0/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_0/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_1/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_1/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/memory_layer/kernel:0, (128, 128), 
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_0/basic_lstm_cell/kernel:0, (384, 512), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_0/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_1/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_1/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/attention/luong_attention/attention_g:0, (), /device:GPU:0
  dynamic_seq2seq/decoder/attention/attention_layer/kernel:0, (256, 128), /device:GPU:0
  dynamic_seq2seq/decoder/output_projection/kernel:0, (128, 9987), 
# creating eval graph ...
  num_layers = 2, num_residual_layers=0
  cell 0  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
  cell 1  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
  cell 0  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
  cell 1  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
# Trainable variables
  embeddings/encoder/embedding_encoder:0, (11178, 128), /device:GPU:0
  embeddings/decoder/embedding_decoder:0, (9987, 128), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_0/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_0/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_1/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_1/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/memory_layer/kernel:0, (128, 128), 
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_0/basic_lstm_cell/kernel:0, (384, 512), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_0/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_1/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_1/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/attention/luong_attention/attention_g:0, (), /device:GPU:0
  dynamic_seq2seq/decoder/attention/attention_layer/kernel:0, (256, 128), /device:GPU:0
  dynamic_seq2seq/decoder/output_projection/kernel:0, (128, 9987), 
# creating infer graph ...
  num_layers = 2, num_residual_layers=0
  cell 0  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
  cell 1  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
  cell 0  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
  cell 1  LSTM, forget_bias=1  DeviceWrapper, device=/gpu:0
# Trainable variables
  embeddings/encoder/embedding_encoder:0, (11178, 128), /device:GPU:0
  embeddings/decoder/embedding_decoder:0, (9987, 128), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_0/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_0/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_1/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/encoder/rnn/multi_rnn_cell/cell_1/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/memory_layer/kernel:0, (128, 128), 
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_0/basic_lstm_cell/kernel:0, (384, 512), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_0/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_1/basic_lstm_cell/kernel:0, (256, 512), /device:GPU:0
  dynamic_seq2seq/decoder/attention/multi_rnn_cell/cell_1/basic_lstm_cell/bias:0, (512,), /device:GPU:0
  dynamic_seq2seq/decoder/attention/luong_attention/attention_g:0, (), /device:GPU:0
  dynamic_seq2seq/decoder/attention/attention_layer/kernel:0, (256, 128), /device:GPU:0
  dynamic_seq2seq/decoder/output_projection/kernel:0, (128, 9987), 
# log_file=/Users/jigyasa/Desktop/nmt_jigyasa/nmt/tmp/nmt_model/log_1528877531
2018-06-13 10:12:11.698764: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
  created train model with fresh parameters, time 0.19s
  created infer model with fresh parameters, time 0.08s
  created eval model with fresh parameters, time 0.09s
Segmentation fault: 11

I am using a tensorflow version more that 1.4 and my operating system is MAC OS .

@kashif @ewilderj @bastings @jhseu @mrry
